components/common/svg_images_components/public_channel_intro_svg.tsx

```diff
@@ -0,0 +1,409 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from "react";
+
+type SvgProps = {
+    width?: number;
+    height?: number;
+};
+
+const PublicChannelIntroSvg = (props: SvgProps) => (
```

Comment:
When I use this svg, i am getting an error

```
Uncaught Error: The `style` prop expects a mapping from style properties to values, not a string. For example, style={{marginRight: spacing + 'em'}} when using JSX.
    in mask (created by PublicChannelIntroSvg)
    in g (created by PublicChannelIntroSvg)
    in svg (created by PublicChannelIntroSvg)
    in PublicChannelIntroSvg (created by ChannelIntroMessage)
    in div (created by ChannelIntroMessage)
    in ChannelIntroMessage (created by Connect(ChannelIntroMessage))
    in Connect(ChannelIntroMessage) (created by PostListRow)
    in PostListRow (created by Connect(PostListRow))
    in Connect(PostListRow)
    in div
    in Unknown (created by DynamicSizeList)
    in div (created by ItemMeasurer)
    in ItemMeasurer (created by DynamicSizeList)
    in div (created by DynamicSizeList)
    in div (created by DynamicSizeList)
    in DynamicSizeList (created by AutoSizer)
    in div (created by AutoSizer)
    in AutoSizer (created by PostList)
    in div (created by PostList)
    in div (created by PostList)
    in div (created by PostList)
    in div (created by PostList)
    in PostList (created by PostList)
    in div (created by PostList)
    in div (created by PostList)
    in div (created by PostList)
    in PostList (created by Connect(PostList))
    in Connect(PostList) (created by Context.Consumer)
    in withRouter(Connect(PostList)) (created by PostView)
    in div (created by PostView)
    in PostView (created by Connect(PostView))
    in Connect(PostView) (created by Context.Consumer)
    in withRouter(Connect(PostView)) (created by DeferredRenderWrapper)
    in DeferredRenderWrapper (created by ChannelView)
    in div (created by ChannelView)
    in ChannelView (created by Connect(ChannelView))
    in Connect(ChannelView) (created by Context.Consumer)
    in withRouter(Connect(ChannelView)) (created by ChannelIdentifierRouter)
    in ChannelIdentifierRouter (created by Connect(ChannelIdentifierRouter))
    in Connect(ChannelIdentifierRouter) (created by Context.Consumer)
    in withRouter(Connect(ChannelIdentifierRouter)) (created by Context.Consumer)
    in Route (created by CenterChannel)
    in Switch (created by CenterChannel)
    in div (created by CenterChannel)
    in div (created by CenterChannel)
    in CenterChannel (created by Connect(CenterChannel))
    in Connect(CenterChannel) (created by Context.Consumer)
    in Route (created by ChannelController)
    in div (created by ChannelController)
    in div (created by ChannelController)
    in ChannelController (created by Context.Consumer)
    in Route (created by NeedsTeam)
    in Switch (created by NeedsTeam)
    in NeedsTeam (created by Connect(NeedsTeam))
    in Connect(NeedsTeam) (created by Context.Consumer)
    in withRouter(Connect(NeedsTeam)) (created by Context.Consumer)
    in LoggedIn (created by Connect(LoggedIn))
    in Connect(LoggedIn) (created by LoggedIn)
    in Suspense (created by LoggedIn)
    in LoggedIn (created by Context.Consumer)
    in Route (created by LoggedInRoute)
    in LoggedInRoute (created by Root)
    in Switch (created by Root)
    in Fe (created by ThemeProvider)
    in ThemeProvider (created by CompassThemeProvider)
    in CompassThemeProvider (created by Root)
    in Switch (created by Root)
    in IntlProvider (created by IntlProvider)
    in IntlProvider (created by Connect(IntlProvider))
    in Connect(IntlProvider) (created by Root)
    in Root (created by Connect(Root))
    in Connect(Root) (created by Root)
    in Suspense (created by Root)
    in Root (created by Context.Consumer)
    in Route (created by App)
    in Router (created by App)
    in Provider (created by App)
    in App
    at assertValidProps (react-dom.development.js?61bb:4760:1)
    at setInitialProperties (react-dom.development.js?61bb:6025:1)
    at finalizeInitialChildren (react-dom.development.js?61bb:7499:1)
    at completeWork (react-dom.development.js?61bb:18978:1)
    at completeUnitOfWork (react-dom.development.js?61bb:22192:1)
    at performUnitOfWork (react-dom.development.js?61bb:22165:1)
    at workLoopSync (react-dom.development.js?61bb:22130:1)
    at performSyncWorkOnRoot (react-dom.development.js?61bb:21756:1)
    at eval (react-dom.development.js?61bb:11089:1)
    at unstable_runWithPriority (scheduler.development.js?3069:653:1)
 ```

Is this an issue of svg or I am doing something wrong? @hmhealey @M-ZubairAhmed

Reply:
I figured out that when I copy paste another svg from the project in the same file, it works. Now, shall I use png?

The svgs are here
https://www.figma.com/file/LHSb4bMcvGles9ASdpA4sN/MM-28014-Channel-Intro-Messages?node-id=522%3A36531

Reply:
I think the `style="mask-type:alpha"` below is what's causing this error since SVGs and HTML expect it to be a string, but React expects it to be an object. You might be able to just change those to `style={{maskType: 'alpha'}}` to fix that.

Reply:
This is a great tool for quickly converting SVGs to React components. Not sure if this will help, but thought I'd share

Reply:
Thanks @matthewbirtch @hmhealey It works fine now :)